### PR TITLE
Fix/add node modules to s3

### PR DESCRIPTION
**Pull request by krampstudio**
Upload also `views/node_modules` files to s3

How to test : 
- configure the file `views/build/config/aws.json`
```
cd views/build
npm i
npx grunt  clean:s3 compress:s3 awsS3:upload
```
- check the bucket contains all the files under `node_modules`

### CODE REVIEW

File: views/build/grunt/s3.js [hunk 1]
```diff
@@ -1,26 +1,56 @@
...
+ *
+ * @author Bertrand Chevrier <bertrand@taotesting.com>
+ *
+ * @param {Object} grunt - the grunt instance
+ */
 module.exports = function(grunt) {
     'use strict';
 
-    var awsConfig  = require('../config/aws.json');
+    const awsConfig  = require('../config/aws.json');
```
Comment (serge-salamanka-1pt) [inline views/build/grunt/s3.js]: the file already exists, probably `config/aws.json` should be renamed to `config/aws.json.template`

the `awsConfig` should be better taken from ENV variables provided with `npx grunt clean:s3 compress:s3 awsS3:upload` command. this is much easier and more secure for automation to be implemented.

Comment (krampstudio) [reply views/build/grunt/s3.js]: what would be the variable names 
`AWS_REGION`, `AWS_ACCESS_KEY`, `AWS_SECRET_KEY`, `AWS_S3_BUCKET`, `AWS_S3_PATH`
(can be accessed by `process.env`)
@zagovorichev can someone from your team take over this ?

**Comment by krampstudio** [comment]
I've made a proposal, @zagovorichev can someone from your team take over this ?

**Comment by zagovorichev** [comment]
yes, sure, I'm working on it, thank you @krampstudio !

File: views/build/grunt/s3.js [hunk 2]
```diff
@@ -1,72 +1,131 @@
...
+ *  AWS_S3_BUCKET,
+ *  AWS_S3_PATH
+ * Or
+ *  copy tao/views/build/config/aws.json.sample to tao/views/build/config/aws.json
+ *  and fill the configuration
+ *
+ * How to upload assets on S3 :
+ *  cd tao/views/build
+ *  npm ci
+ *  npx grunt clean:s3 compress:s3 awsS3:upload
```
Comment (zagovorichev) [inline views/build/grunt/s3.js]: can you change, please `aws_s3`

File: views/build/grunt/s3.js [hunk 3]
```diff
@@ -1,72 +1,131 @@
...
+ *
+ * The upload concurrency can be changed using
+ *  npx grunt clean:s3 compress:s3 awsS3:upload --s3-concurrency=N
```
Comment (zagovorichev) [inline views/build/grunt/s3.js]: can you change, please `aws_s3`

File: views/build/grunt/s3.js [hunk 4]
```diff
@@ -1,72 +1,131 @@
...
-            expand: true,
-            cwd: root,
-            src: patterns,
-            dest : awsConfig.s3.path
-        }]
-    };
+        },
 
-    grunt.loadNpmTasks('grunt-aws-s3');
+        awsS3 : {
```
Comment (zagovorichev) [inline views/build/grunt/s3.js]: can you change, please `aws_s3`

**Comment by zagovorichev** [comment]
Tested on the `ops-act-update-static` server, works with .json with configuration and with ENV parameters

**Pull request merged**